Automattic/jetpack
# Sync: Ensure filtering orders by status when doing HPOS Checksums

Very similar to #38688
## Proposed changes:

* Changed `get_all_possible_order_status_keys` to static and public to filter status in the Checksums

### Other information:


## Jetpack product discussion


p9dueE-8ec-p2#comment-10544
## Does this pull request change what data or activity we track or use?



## Testing instructions:





Start with a Jetpack connected site with WooCommerce, HPOS enabled, and a few orders. Follow the pre-requisites in D154724-code to run a checksum.
- Then, before applying this PR just change one order directly in the DB in your testing site so that its `status` is not one of the allowed ones. Example change `wc-processing` for `processing`. (Anything should work, but this is one of the cases I found failing in a real scenario.)
- Try running a checksum fix again for `wc_orders`as before. There will be differences in the result for that specific order id. No matter how many times the checksum fix is run, the difference will not disappear.
- Apply the patch
- Run again the checksum fix for `wc_orders`. The order id will show now as missing in remote. Makes sense since for checksum purposes we don't include orders with invalid `status` keys now. The checksum fix will actually remove the order from the cache site, since remote is the source of truth. If run again, no differences or missing orders will be found.
- Now let's revert the change in the test site DB so that the order has a valid `status` key.
- Run once more the checksum fix for `wc_orders`. Now it will show that it is missing on wpcom. It makes sense since we removed it from the cached site. The fix will add it to wpcom. 
- Run one last time the checksum fix. No differences or missing orders will be found.
